score/etl/sources/census/etl.py

```diff
@@ -0,0 +1,115 @@
+import csv
+import os
+import json
+from pathlib import Path
+import logging
+
+from .etl_utils import get_state_fips_codes
+from utils import unzip_file_from_url
+
+
+def download_census_csvs(data_path: Path) -> None:
+    # the fips_states_2010.csv is generated from data here
+    # https://www.census.gov/geographies/reference-files/time-series/geo/tallies.html
+    state_fips_codes = get_state_fips_codes(data_path)
+    for fips in state_fips_codes:
+        # check if file exists
+        shp_file_path = data_path.joinpath(
+            "census", "shp", fips, f"tl_2010_{fips}_bg10.shp"
+        )
+        if not os.path.isfile(shp_file_path):
+            logging.info(f"Downloading {fips}")
+
+            # 2020 tiger data is here: https://www2.census.gov/geo/tiger/TIGER2020/BG/
+            # But using 2010 for now
+            cbg_state_url = f"https://www2.census.gov/geo/tiger/TIGER2010/BG/2010/tl_2010_{fips}_bg10.zip"
+            unzip_file_from_url(
+                cbg_state_url, data_path / "tmp", data_path / "census" / "shp" / fips
+            )
+
+        geojson_dir_path = data_path.joinpath(
+            "census",
+            "geojson",
+        )
+
+        if not os.path.isfile(geojson_dir_path.joinpath(fips + ".json")):
+            # ogr2ogr
+            logging.info(f"Encoding GeoJSON for {fips}")
+
+            # if it's running inside Docker container
+            if os.environ.get("DOCKER_CONTAINER"):
+                # TODO
+                pass
+            else:
+                # this is being run on virtualenv
+                # PWD is different for Windows
+                if os.name == "nt":
+                    pwd = "%cd%"
+                else:
+                    pwd = "${PWD}"
+                cmd = (
+                    'docker run --rm -it -v "'
+                    + pwd
+                    + '"/:/home osgeo/gdal:alpine-ultrasmall-latest ogr2ogr -f GeoJSON /home/data/census/geojson/'
+                    + fips
+                    + ".json /home/data/census/shp/"
+                    + fips
+                    + "/tl_2010_"
+                    + fips
+                    + "_bg10.shp"
+                )
+                os.system(cmd)
+
+        # generate CBG CSV table for pandas
+        ## load in memory
+        cbg_national_list = []  # in-memory global list
+        cbg_per_state_list: dict = {}  # in-memory dict per state
```

Comment:
nit: why is this variable named `_list` if it's a `dict`?

Reply:
Good catch. Fixed.